woo-razorpay.php

```diff
@@ -1824,6 +1849,140 @@ public function update1ccOrderWC(& $order, $wcOrderId, $razorpayPaymentId)
             $order->add_order_note( $note );
         }
 
+        public function updateGiftCardData($razorpayData, $order, $orderId, $razorpayPaymentId)
+        {
+            global $woocommerce;
+            global $wpdb;
+
+            foreach($razorpayData['promotions'] as $giftcard)
+            {
+                if($giftcard['type'] == 'gift_card'){
+
+                    $usedAmt = $giftcard['value']/100;
+                    $giftCode = $giftcard['code'];
+                    if(is_plugin_active('yith-woocommerce-gift-cards/init.php')){
+
+                        $yithCard = new YITH_YWGC_Gift_Card( $args = array('gift_card_number'=> $giftCode));
+
+                        $giftCardBalance = $yithCard->get_balance();
+
+                        if($giftCardBalance == null && $balance >= 0 && $usedAmt > $giftCardBalance ){
+                            // initiate refund in case gift card faliure
+                            $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                        }else{
+                            //Deduct amount of gift card
+                            $yithCard->update_balance( $yithCard->get_balance() - $usedAmt );
+                            $yithCard->register_order($orderId);
+
+                            $code = "Gift Card ($giftCode)";
+
+                            $wpdb->query( // phpcs:ignore
+                                $wpdb->prepare(
+                                    'INSERT INTO `' . $wpdb->prefix . 'woocommerce_order_items`( order_item_name, order_item_type, order_id ) VALUES ( %s, %s, %s)',
+                                    $code,
+                                    'fee',
+                                    $orderId
+                                )
+                            );
+
+                            $itemId = $wpdb->insert_id;
+                            wc_add_order_item_meta( $itemId, '_line_total', -$usedAmt );
+                        }
+
+                    }
+                    else if(is_plugin_active('pw-woocommerce-gift-cards/pw-gift-cards.php'))
```

Comment:
What if the merchant is installed and activated both GC plugins? How this logic react?

Reply:
Customer can use only one plugin at a time